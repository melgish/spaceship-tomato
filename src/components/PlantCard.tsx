import type { Plant } from '@/actions/findPlants'
import PHRange from './PHRange'
import RemovePlantButton from './RemovePlantButton'

export default function PlantCard(row: Readonly<Plant>) {
  return (
    <div className="kv-card">
      <div>Name</div>
      <div>
        {row.name} <RemovePlantButton row={row} />
      </div>
      <div>PH</div>
      <div>
        <PHRange range={row} />
      </div>
    </div>
  )
}
