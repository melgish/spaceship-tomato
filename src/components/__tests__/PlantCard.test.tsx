import { expect, it } from 'vitest'
import { render } from '@testing-library/react'
import PlantCard from '../PlantCard'

// A typical Plant.
const data = {
  id: 1,
  name: 'Test Plant',
  minPH: 6,
  maxPH: 7,
}

it('should display the supplied plant', () => {
  const { container } = render(<PlantCard {...data} />)

  expect(container).toMatchSnapshot()
})
