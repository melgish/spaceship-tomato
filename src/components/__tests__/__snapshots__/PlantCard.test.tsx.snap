// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should display the supplied plant 1`] = `
<div>
  <div
    class="kv-card"
  >
    <div>
      Name
    </div>
    <div>
      Test Plant
       
      <button
        class="px-1 py-0 text-red-800 font-bold border-red-800 text-sm"
        type="button"
      >
        Remove
      </button>
    </div>
    <div>
      PH
    </div>
    <div>
      6.0
       â€“ 
      7.0
    </div>
  </div>
</div>
`;
